app/scripts/controller-init/account-tracker-controller-init.ts

```diff
@@ -1,56 +1,38 @@
-import { assert } from '@metamask/utils';
-import { getProviderConfig } from '../../../shared/modules/selectors/networks';
-import { NETWORK_TYPES } from '../../../shared/constants/network';
-import AccountTrackerController from '../controllers/account-tracker-controller';
+import { AccountTrackerController } from '@metamask/assets-controllers';
+import { NetworkClientId } from '@metamask/network-controller';
 import { ControllerInitFunction } from './types';
 import {
   AccountTrackerControllerInitMessenger,
   AccountTrackerControllerMessenger,
 } from './messengers';
 
-/**
- * Initialize the account tracker controller.
- *
- * @param request - The request object.
- * @param request.controllerMessenger - The messenger to use for the controller.
- * @param request.initMessenger - The messenger to use for initialization.
- * @returns The initialized controller.
- */
 export const AccountTrackerControllerInit: ControllerInitFunction<
   AccountTrackerController,
   AccountTrackerControllerMessenger,
   AccountTrackerControllerInitMessenger
-> = ({ controllerMessenger, initMessenger }) => {
-  const { provider, blockTracker } =
-    initMessenger.call('NetworkController:getSelectedNetworkClient') ?? {};
-
-  assert(
-    provider,
-    'Provider is required to initialize AccountTrackerController.',
-  );
-
-  assert(
-    blockTracker,
-    'Block tracker is required to initialize AccountTrackerController.',
-  );
+> = ({ controllerMessenger, initMessenger, persistedState, getController }) => {
+  const getAssetsContractController = () =>
+    getController('AssetsContractController');
 
   const controller = new AccountTrackerController({
-    state: { accounts: {} },
+    state: persistedState.AccountTrackerController,
     messenger: controllerMessenger,
-    provider,
-    blockTracker,
-    getNetworkIdentifier: (providerConfig): string => {
-      const metamask = initMessenger.call('NetworkController:getState');
-
-      const config =
-        providerConfig ??
-        getProviderConfig({
-          metamask,
-        });
-
-      return config.type === NETWORK_TYPES.RPC && config.rpcUrl
-        ? config.rpcUrl
-        : config.type;
+    getStakedBalanceForChain: (
+      addresses: string[],
+      networkClientId?: NetworkClientId,
+    ) => {
+      const assetsContractController = getAssetsContractController();
+      return assetsContractController.getStakedBalanceForChain(
+        addresses,
+        networkClientId,
+      );
+    },
+    includeStakedAssets: false,
```

Comment:
Not part of this PR to include staked assets in the assets list. When we want to enable them, we can do so.